### PR TITLE
Use merge commit sha and mergeable on PRs to determine if the test merge is ready to run

### PR DESCRIPTION
**Pull request by luck3y**

### CODE REVIEW

File: src/main/java/org/jboss/pull/player/PullPlayer.java [hunk 1]
```diff
@@ -76,21 +77,38 @@ private void processPulls(PersistentList whiteList, PersistentList adminList, Li
...
+                continue;
+            }
+            // this is the test commit created that is a merge onto the branch the pr has been opened against.
+            // see: https://developer.github.com/v3/pulls/#response-1
+            String mergeCommitSha = prDetails.get("merge_commit_sha").asString();
+            if (mergeCommitSha == null) {
+                System.out.println("No vaild merge_commit_sha found on PR, skipping.");
+                gitHubApi.postComment(pullNumber, "There was no valid merge_commit_sha found on this PR though mergeable is set to true, it will be retried.");
+                continue;
+            }
```
Comment (bstansberry) [inline src/main/java/org/jboss/pull/player/PullPlayer.java:107]: I'm not sure all the processing below should be skipped in the mergeable = false (as opposed to null) case. IOW if the PR has a merge conflict the pull player won't respond to anything, but in some cases perhaps it should

Comment (luck3y) [reply src/main/java/org/jboss/pull/player/PullPlayer.java:107]: Ah, yeah good point, I'll make this more selective so commands still work, thanks!

Comment (bstansberry) [reply src/main/java/org/jboss/pull/player/PullPlayer.java:107]: But probably not worth the complexity of dealing with that for a fairly understandable situation -- GH tells you if you have a conflict so it's not hard to understand why the pull player might not be doing something.

Sp LGTM.

Comment (luck3y) [reply src/main/java/org/jboss/pull/player/PullPlayer.java:107]: Its a pretty easy fix to just delay the check until after the command loop, so i just pushed that.

File: src/main/java/org/jboss/pull/player/Main.java [hunk 2]
```diff
@@ -5,6 +5,8 @@
  */
 package org.jboss.pull.player;
 
+import org.jboss.dmr.ModelNode;
```
Comment (bstansberry) [inline src/main/java/org/jboss/pull/player/Main.java]: This looks to be cruft.

Comment (luck3y) [reply src/main/java/org/jboss/pull/player/Main.java]: +1 will do a sweep for these as well. Note I'm pretty much done with testing this (I tested on jboss-openshift-modules, so we can push it out and keep an eye on it in the morning.)

**Pull request merged**